src/platformdirs/macos.py

```diff
@@ -0,0 +1,64 @@
+import os
+
+from .api import PlatformDirsABC
+
+
+class MacOS(PlatformDirsABC):
+    """
+    Platform directories for the macOS operating system. Follows the guidance from `Apple documentation
+    <https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html>`_.
+    Makes use of the `appname <platformdirs.api.PlatformDirsABC.appname>` and
+    `version <platformdirs.api.PlatformDirsABC.version>`.
+    """
+
+    @property
+    def user_data_dir(self) -> str:
+        """:return: data directory tied to the user, e.g. ``~/Library/Application Support/$appname/$version``"""
+        return self._path_with_app_name_version("~/Library/Application Support/", expand=True)
+
+    def _path_with_app_name_version(self, of: str, *, expand: bool = False) -> str:
```

Comment:
would it make sense to provide this as helper in the base class, at first glance im under the impression this logic is always the same

i would still recommend to take the expand out, and make it `*of : str`

reason is that expand is a bool trap, and callers can just layer it in here as its no longer public

Reply:
Addressed.